### ISSUE TITLE
NPE thrown when trying to update entire extension resource (i.e. EnterpriseExtension)

### ISSUE THREAD

**Issue by jasonfagerberg-toast** [apache/directory-scimple#695]
When provided a patch operation like so

```json
{
  "schemas": ["urn:ietf:params:scim:api:messages:2.0:PatchOp"],
  "Operations": [
    {
      "op": "replace",
      "path": "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User",
      "value": {
        "costCenter": "New Cost Center",
        "department": "New Department"
      }
    }
  ]
}
```

`DefaultPatchHander` throws a Null Pointer Exception

```
java.lang.NullPointerException
	at org.apache.directory.scim.core.repository.DefaultPatchHandler.apply(DefaultPatchHandler.java:135)
	at org.apache.directory.scim.core.repository.DefaultPatchHandler.apply(DefaultPatchHandler.java:118)
	at org.apache.directory.scim.core.repository.PatchHandlerTest.applyReplaceEntireEnterpriseExtension(PatchHandlerTest.java:327)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
```

This seems to be because `DefaultPatchHandler.attributeReference` incorrectly parses the path as though it is an attribute path instead of a value path

![image](https://github.com/user-attachments/assets/44638688-3449-4266-87c4-fd43d4beeb02)

I believe by reading [RFC-7644](https://datatracker.ietf.org/doc/html/rfc7644#section-3.5.2.3) this should be a valid request. Please feel free to correct me if I am wrong! 😄 

I have written a test to illustrate the error https://github.com/apache/directory-scimple/pull/694

**Comment by jasonfagerberg-toast** [comment]
```
{
  "schemas": ["urn:ietf:params:scim:api:messages:2.0:PatchOp"],
  "Operations": [
    {
      "op": "replace",
      "path": "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User",
      "value": {
        "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User": {
          "costCenter": "New Cost Center",
          "department": "New Department"
        }
      }
    }
  ]
}
```

also fails for the same reason

**Comment by bdemers** [comment]
Hey @jasonfagerberg-toast thanks for reporting and digging into this!
The context provided and the added test make it really easy to understand the issue.

My guess is this is just a missing functionality in the PatchHandler.  Based on your test i'll try to take a look and get a fix in.

Again, thanks for the time/effort you put into this bug report!

**Comment by bdemers** [comment]
Fixed in #694
